src/main/java/kg/jarkyn/cobspecserver/ClientSocket.java

```diff
@@ -0,0 +1,28 @@
+package kg.jarkyn.cobspecserver;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+
+public class ClientSocket implements Client {
```

Comment:
Good catch, CJ.
I think the separation between `Client` and `ClientSocket` is not needed here and could be seen as an overkill. Personally I'd get rid of the interface `Client` and only have a `ClientSocket` class in the application.